### PR TITLE
Fix casing of files to get closer to ./build.sh -allConfiguration working on linux

### PR DESCRIPTION
**Pull request by krwq**
Currently `./build.sh -allConfiguration` is not working on linux.

This is fixing one class of issues.
There are still some errors I'm seeing locally but not sure how to fix yet (will be digging a bit more):

```
/home/krwq/.nuget/packages/microsoft.dotnet.build.tasks.packaging/1.0.0-beta.19081.1/build/Packaging.targets(1255,5): error : Error when creating nuget packed package from /tmp/tmp.A9Djs3tsi9/artifacts/packages/Debug/specs/runtime.linux-x64-aot.Microsoft.Private.CoreFx.NETCoreApp.nuspec. System.IO.DirectoryNotFoundException: Could not find a part of the path '/tmp/tmp.A9Djs3tsi9/artifacts/bin/native/netcoreapp-Linux-Debug-x64-aot'. [/tmp/tmp.A9Djs3tsi9/pkg/Microsoft.Private.CoreFx.NETCoreApp/Microsoft.Private.CoreFx.NETCoreApp.pkgproj]
/home/krwq/.nuget/packages/microsoft.dotnet.build.tasks.packaging/1.0.0-beta.19081.1/build/Packaging.targets(1255,5): error :    at System.IO.Enumeration.FileSystemEnumerator`1.CreateDirectoryHandle(String path, Boolean ignoreNotFound) [/tmp/tmp.A9Djs3tsi9/pkg/Microsoft.Private.CoreFx.NETCoreApp/Microsoft.Private.CoreFx.NETCoreApp.pkgproj]
/home/krwq/.nuget/packages/microsoft.dotnet.build.tasks.packaging/1.0.0-beta.19081.1/build/Packaging.targets(1255,5): error :    at System.IO.Enumeration.FileSystemEnumerator`1..ctor(String directory, EnumerationOptions options) [/tmp/tmp.A9Djs3tsi9/pkg/Microsoft.Private.CoreFx.NETCoreApp/Microsoft.Private.CoreFx.NETCoreApp.pkgproj]
/home/krwq/.nuget/packages/microsoft.dotnet.build.tasks.packaging/1.0.0-beta.19081.1/build/Packaging.targets(1255,5): error :    at System.IO.Enumeration.FileSystemEnumerable`1..ctor(String directory, FindTransform transform, EnumerationOptions options) [/tmp/tmp.A9Djs3tsi9/pkg/Microsoft.Private.CoreFx.NETCoreApp/Microsoft.Private.CoreFx.NETCoreApp.pkgproj]
/home/krwq/.nuget/packages/microsoft.dotnet.build.tasks.packaging/1.0.0-beta.19081.1/build/Packaging.targets(1255,5): error :    at System.IO.Enumeration.FileSystemEnumerableFactory.UserFiles(String directory, String expression, EnumerationOptions options) [/tmp/tmp.A9Djs3tsi9/pkg/Microsoft.Private.CoreFx.NETCoreApp/Microsoft.Private.CoreFx.NETCoreApp.pkgproj]
/home/krwq/.nuget/packages/microsoft.dotnet.build.tasks.packaging/1.0.0-beta.19081.1/build/Packaging.targets(1255,5): error :    at System.IO.Directory.InternalEnumeratePaths(String path, String searchPattern, SearchTarget searchTarget, EnumerationOptions options) [/tmp/tmp.A9Djs3tsi9/pkg/Microsoft.Private.CoreFx.NETCoreApp/Microsoft.Private.CoreFx.NETCoreApp.pkgproj]
/home/krwq/.nuget/packages/microsoft.dotnet.build.tasks.packaging/1.0.0-beta.19081.1/build/Packaging.targets(1255,5): error :    at System.IO.Directory.GetFiles(String path, String searchPattern, SearchOption searchOption) [/tmp/tmp.A9Djs3tsi9/pkg/Microsoft.Private.CoreFx.NETCoreApp/Microsoft.Private.CoreFx.NETCoreApp.pkgproj]
/home/krwq/.nuget/packages/microsoft.dotnet.build.tasks.packaging/1.0.0-beta.19081.1/build/Packaging.targets(1255,5): error :    at NuGet.Common.PathResolver.PerformWildcardSearch(String basePath, String searchPath, Boolean includeEmptyDirectories, String& normalizedBasePath) [/tmp/tmp.A9Djs3tsi9/pkg/Microsoft.Private.CoreFx.NETCoreApp/Microsoft.Private.CoreFx.NETCoreApp.pkgproj]
/home/krwq/.nuget/packages/microsoft.dotnet.build.tasks.packaging/1.0.0-beta.19081.1/build/Packaging.targets(1255,5): error :    at NuGet.Packaging.PackageBuilder.ResolveSearchPattern(String basePath, String searchPath, String targetPath, Boolean includeEmptyDirectories) [/tmp/tmp.A9Djs3tsi9/pkg/Microsoft.Private.CoreFx.NETCoreApp/Microsoft.Private.CoreFx.NETCoreApp.pkgproj]
/home/krwq/.nuget/packages/microsoft.dotnet.build.tasks.packaging/1.0.0-beta.19081.1/build/Packaging.targets(1255,5): error :    at NuGet.Packaging.PackageBuilder.AddFiles(String basePath, String source, String destination, String exclude) [/tmp/tmp.A9Djs3tsi9/pkg/Microsoft.Private.CoreFx.NETCoreApp/Microsoft.Private.CoreFx.NETCoreApp.pkgproj]
/home/krwq/.nuget/packages/microsoft.dotnet.build.tasks.packaging/1.0.0-beta.19081.1/build/Packaging.targets(1255,5): error :    at NuGet.Packaging.PackageBuilder.PopulateFiles(String basePath, IEnumerable`1 files) [/tmp/tmp.A9Djs3tsi9/pkg/Microsoft.Private.CoreFx.NETCoreApp/Microsoft.Private.CoreFx.NETCoreApp.pkgproj]
/home/krwq/.nuget/packages/microsoft.dotnet.build.tasks.packaging/1.0.0-beta.19081.1/build/Packaging.targets(1255,5): error :    at Microsoft.DotNet.Build.Tasks.Packaging.NuGetPack.Pack(String nuspecPath, String nupkgPath, Manifest manifest, Boolean packSymbols) in /_/src/Microsoft.DotNet.Build.Tasks.Packaging/src/NuGetPack.cs:line 256 [/tmp/tmp.A9Djs3tsi9/pkg/Microsoft.Private.CoreFx.NETCoreApp/Microsoft.Private.CoreFx.NETCoreApp.pkgproj]
/home/krwq/.nuget/packages/microsoft.dotnet.build.tasks.packaging/1.0.0-beta.19081.1/build/Packaging.targets(1255,5): error : Error when creating nuget packed package from /tmp/tmp.A9Djs3tsi9/artifacts/packages/Debug/specs/runtime.linux-x64-aot.Microsoft.Private.CoreFx.NETCoreApp.nuspec. System.IO.DirectoryNotFoundException: Could not find a part of the path '/tmp/tmp.A9Djs3tsi9/artifacts/bin/native/netcoreapp-Linux-Debug-x64-aot'. [/tmp/tmp.A9Djs3tsi9/pkg/Microsoft.Private.CoreFx.NETCoreApp/Microsoft.Private.CoreFx.NETCoreApp.pkgproj]
/home/krwq/.nuget/packages/microsoft.dotnet.build.tasks.packaging/1.0.0-beta.19081.1/build/Packaging.targets(1255,5): error :    at System.IO.Enumeration.FileSystemEnumerator`1.CreateDirectoryHandle(String path, Boolean ignoreNotFound) [/tmp/tmp.A9Djs3tsi9/pkg/Microsoft.Private.CoreFx.NETCoreApp/Microsoft.Private.CoreFx.NETCoreApp.pkgproj]
/home/krwq/.nuget/packages/microsoft.dotnet.build.tasks.packaging/1.0.0-beta.19081.1/build/Packaging.targets(1255,5): error :    at System.IO.Enumeration.FileSystemEnumerator`1..ctor(String directory, EnumerationOptions options) [/tmp/tmp.A9Djs3tsi9/pkg/Microsoft.Private.CoreFx.NETCoreApp/Microsoft.Private.CoreFx.NETCoreApp.pkgproj]
/home/krwq/.nuget/packages/microsoft.dotnet.build.tasks.packaging/1.0.0-beta.19081.1/build/Packaging.targets(1255,5): error :    at System.IO.Enumeration.FileSystemEnumerable`1..ctor(String directory, FindTransform transform, EnumerationOptions options) [/tmp/tmp.A9Djs3tsi9/pkg/Microsoft.Private.CoreFx.NETCoreApp/Microsoft.Private.CoreFx.NETCoreApp.pkgproj]
/home/krwq/.nuget/packages/microsoft.dotnet.build.tasks.packaging/1.0.0-beta.19081.1/build/Packaging.targets(1255,5): error :    at System.IO.Enumeration.FileSystemEnumerableFactory.UserFiles(String directory, String expression, EnumerationOptions options) [/tmp/tmp.A9Djs3tsi9/pkg/Microsoft.Private.CoreFx.NETCoreApp/Microsoft.Private.CoreFx.NETCoreApp.pkgproj]
/home/krwq/.nuget/packages/microsoft.dotnet.build.tasks.packaging/1.0.0-beta.19081.1/build/Packaging.targets(1255,5): error :    at System.IO.Directory.InternalEnumeratePaths(String path, String searchPattern, SearchTarget searchTarget, EnumerationOptions options) [/tmp/tmp.A9Djs3tsi9/pkg/Microsoft.Private.CoreFx.NETCoreApp/Microsoft.Private.CoreFx.NETCoreApp.pkgproj]
/home/krwq/.nuget/packages/microsoft.dotnet.build.tasks.packaging/1.0.0-beta.19081.1/build/Packaging.targets(1255,5): error :    at System.IO.Directory.GetFiles(String path, String searchPattern, SearchOption searchOption) [/tmp/tmp.A9Djs3tsi9/pkg/Microsoft.Private.CoreFx.NETCoreApp/Microsoft.Private.CoreFx.NETCoreApp.pkgproj]
/home/krwq/.nuget/packages/microsoft.dotnet.build.tasks.packaging/1.0.0-beta.19081.1/build/Packaging.targets(1255,5): error :    at NuGet.Common.PathResolver.PerformWildcardSearch(String basePath, String searchPath, Boolean includeEmptyDirectories, String& normalizedBasePath) [/tmp/tmp.A9Djs3tsi9/pkg/Microsoft.Private.CoreFx.NETCoreApp/Microsoft.Private.CoreFx.NETCoreApp.pkgproj]
/home/krwq/.nuget/packages/microsoft.dotnet.build.tasks.packaging/1.0.0-beta.19081.1/build/Packaging.targets(1255,5): error :    at NuGet.Packaging.PackageBuilder.ResolveSearchPattern(String basePath, String searchPath, String targetPath, Boolean includeEmptyDirectories) [/tmp/tmp.A9Djs3tsi9/pkg/Microsoft.Private.CoreFx.NETCoreApp/Microsoft.Private.CoreFx.NETCoreApp.pkgproj]
/home/krwq/.nuget/packages/microsoft.dotnet.build.tasks.packaging/1.0.0-beta.19081.1/build/Packaging.targets(1255,5): error :    at NuGet.Packaging.PackageBuilder.AddFiles(String basePath, String source, String destination, String exclude) [/tmp/tmp.A9Djs3tsi9/pkg/Microsoft.Private.CoreFx.NETCoreApp/Microsoft.Private.CoreFx.NETCoreApp.pkgproj]
/home/krwq/.nuget/packages/microsoft.dotnet.build.tasks.packaging/1.0.0-beta.19081.1/build/Packaging.targets(1255,5): error :    at NuGet.Packaging.PackageBuilder.PopulateFiles(String basePath, IEnumerable`1 files) [/tmp/tmp.A9Djs3tsi9/pkg/Microsoft.Private.CoreFx.NETCoreApp/Microsoft.Private.CoreFx.NETCoreApp.pkgproj]
/home/krwq/.nuget/packages/microsoft.dotnet.build.tasks.packaging/1.0.0-beta.19081.1/build/Packaging.targets(1255,5): error :    at Microsoft.DotNet.Build.Tasks.Packaging.NuGetPack.Pack(String nuspecPath, String nupkgPath, Manifest manifest, Boolean packSymbols) in /_/src/Microsoft.DotNet.Build.Tasks.Packaging/src/NuGetPack.cs:line 256 [/tmp/tmp.A9Djs3tsi9/pkg/Microsoft.Private.CoreFx.NETCoreApp/Microsoft.Private.CoreFx.NETCoreApp.pkgproj]
```

Not seeing `netcoreapp-Linux-Debug-x64-aot` in my bin folder

```
/tmp/tmp.A9Djs3tsi9/artifacts/bin/native$ ls -la
total 20
drwxr-xr-x   3 krwq krwq  4096 Feb  4 12:05 .
drwxr-xr-x 207 krwq krwq 12288 Feb  4 13:03 ..
drwxr-xr-x   2 krwq krwq  4096 Feb  4 12:55 netcoreapp-Linux-Debug-x64
```

cc: @dotnet/dnceng

### CODE REVIEW

**Comment by dagood** [comment]
I was working on this (linux allconfigurations) for source-build! I summarized my latest progress at https://github.com/dotnet/source-build/issues/210#issuecomment-454528533 because I had to stop working on it.

The `aot` issue happens when you try to build the runtime packages while also trying to build allconfigurations:

> Initially I tried to use a single command to build `-allconfigurations` and the vertical for the current platform within the same build context. That has complications ([dotnet/corefx#34332 (review)](https://github.com/dotnet/corefx/pull/34332#pullrequestreview-189954566)) so I abandoned it and instead first execute a build command for `-allconfigurations`, then another build command for the vertical.

/cc @dleeapho @dseefeld @crummel

**Comment by krwq** [comment]
@dagood hope this is not conflicting with anything. In the issue you mention you got the workaround for the build.. I just need to be able to build packages locally. You mentioned to use `-allconfigurations` Which verticals do I need to build? (and how?)

**Comment by dagood** [comment]
No conflict. It's cool to see someone else trying to do this. 😄

> I just need to be able to build packages locally.

Depending on which packages you want to build, you don't need `-allconfigurations`. Since you're trying to use allconfigurations, I assume you've already done a build and it didn't produce the specific packages you wanted, but I want to point that out to make sure.

> Which verticals do I need to build? (and how?)

I think it would take too long for me to go back and try to figure out what the minimum is to do this. It has likely been broken by later corefx commits, too. You can try taking my WIP branch of source-build (https://github.com/dagood/source-build/tree/corefx-all), building it, and checking the state of the `src/corefx` submodule after patches and the build commands it tells you it used for corefx and corefx-allconfigurations.

File: src/Microsoft.IO.Redist/src/Microsoft.IO.Redist.csproj [hunk 1]
```diff
@@ -128,7 +128,7 @@
     <Compile Include="$(CommonPath)\Interop\Windows\kernel32\Interop.DeleteVolumeMountPoint.cs">
       <Link>Common\Interop\Windows\Interop.DeleteVolumeMountPoint.cs</Link>
     </Compile>
-    <Compile Include="$(CommonPath)\CoreLib\Interop\Windows\kernel32\Interop.GetLogicalDrives.cs">
+    <Compile Include="$(CommonPath)\CoreLib\Interop\Windows\Kernel32\Interop.GetLogicalDrives.cs">
```
Comment (safern) [inline src/Microsoft.IO.Redist/src/Microsoft.IO.Redist.csproj]: Should we instead be consistent and rename this directories to be all lowercase? It is odd to have someones to be lowercase and other uppercase.

File: src/System.Diagnostics.PerformanceCounter/src/System.Diagnostics.PerformanceCounter.csproj [hunk 2]
```diff
@@ -47,11 +47,11 @@
     <Compile Include="$(CommonPath)\CoreLib\Interop\Windows\Kernel32\Interop.FormatMessage.cs">
       <Link>Common\CoreLib\Interop\Windows\Interop.FormatMessage.cs</Link>
     </Compile>
-    <Compile Include="$(CommonPath)\CoreLib\Interop\Windows\advapi32\Interop.RegCloseKey.cs">
+    <Compile Include="$(CommonPath)\CoreLib\Interop\Windows\Advapi32\Interop.RegCloseKey.cs">
```
Comment (safern) [inline src/System.Diagnostics.PerformanceCounter/src/System.Diagnostics.PerformanceCounter.csproj]: Same here.

Comment (krwq) [reply src/System.Diagnostics.PerformanceCounter/src/System.Diagnostics.PerformanceCounter.csproj]: which do we prefer?

Comment (safern) [reply src/System.Diagnostics.PerformanceCounter/src/System.Diagnostics.PerformanceCounter.csproj]: I would prefer lowercase, and that seem to be the majority of the cases, but I don't know what other people think on this.

cc @jkotas @stephentoub @danmosemsft

Comment (jkotas) [reply src/System.Diagnostics.PerformanceCounter/src/System.Diagnostics.PerformanceCounter.csproj]: The casing of the filename should match the casing of the `class` that the file or directory contains.

The interop classes start with upper case: `class Kernel32`, `class Advapi32`, ... .

Comment (jkotas) [reply src/System.Diagnostics.PerformanceCounter/src/System.Diagnostics.PerformanceCounter.csproj]: The upper case is much more common, e.g.:

`Kernel32` 1571 hits over 415 files
`kernel32` 616 hits over 96 files

Comment (stephentoub) [reply src/System.Diagnostics.PerformanceCounter/src/System.Diagnostics.PerformanceCounter.csproj]: We should use the upper case.

File: src/System.Windows.Extensions/src/System.Windows.Extensions.csproj [hunk 3]
```diff
@@ -27,8 +27,8 @@
     <Compile Include="$(CommonPath)\Interop\Windows\CryptUI\Interop.CryptUIDlgCertificate.cs">
       <Link>Common\Interop\Windows\Crypt32\Interop.CryptUIDlgCertificate.cs</Link>
     </Compile>
-    <Compile Include="$(CommonPath)\Interop\Windows\User32\Interop.MessageBeep.cs">
-      <Link>Common\Interop\Windows\User32\Interop.MessageBeep.cs</Link>
+    <Compile Include="$(CommonPath)\Interop\Windows\user32\Interop.MessageBeep.cs">
+      <Link>Common\Interop\Windows\user32\Interop.MessageBeep.cs</Link>
```
Comment (stephentoub) [inline src/System.Windows.Extensions/src/System.Windows.Extensions.csproj]: We should make these upper case.  If it's not on the file system, we should rename the directory as well.

**Comment by krwq** [comment]
I've also normalized slashed to use Windows, please let me know if you find any other directories which need renaming. Currently testing this locally if fixes original issue

**Comment by krwq** [comment]
@safern ping - this PR is super merge conflict prone

**Comment by krwq** [comment]
Let me know if you got any other massive casing changes, I wrote tiny hacky tool for that so it will go quickly in the future

**Pull request merged**